[storekit] Update for Xcode 10.2 beta 2

VincentDondain opened the pull request:
Also obsoleted incorrect enum value that droped "CloudService" by mistake.

monojenkins commented:
Build success
✅ [Build succeeded](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/9196//console)
✅ [API Diff (from stable)](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/9196//API_20diff_20_28from_20stable_29)
⚠️ [API Diff (from PR only)](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/9196//API_20diff_20_28PR_20only_29) (🔥 breaking changes 🔥)
ℹ️ [Generator Diff](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/9196//Generator_20Diff) (please review changes)
✅ [Test run succeeded](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/9196//Test_20Report)

VincentDondain commented:
API diff errors are legit but those enum values haven't shipped so it's fine (:

```
Namespace StoreKit

Type Changed: StoreKit.SKError

Removed values:

	InvalidOfferIdentifier = 9,
	InvalidOfferPrice = 12,
	InvalidSignature = 10,
	MissingOfferParams = 11,
Modified fields:

	PrivacyAcknowledgementRequired = 13 9
```
